Fix doji candlesticks in webgl

tradingcage opened the pull request:
Currently, when you use a WebGL series to display a pure doji candlestick (where the close and open price are equal), it sets the candlestick body height to 0, omitting it entirely. This looks weird because it displays the candlesticks as wicks only. This bug doesn't duplicate for canvas candlestick series, only webgl. This PR fixes it by setting the height of the body to the minimum value when the open and close are equal.

Tested by editing examples/simple-chart/index.js to look like this:

```
const data = fc.randomFinancial()(50).map(bar => ({ ...bar, close: bar.open }));

const yExtent = fc.extentLinear().accessors([d => d.high, d => d.low]);

const xExtent = fc.extentDate().accessors([d => d.date]);

const gridlines = fc.annotationSvgGridline();
const candlestick = fc.seriesWebglCandlestick();
const multi = fc.seriesSvgMulti().series([gridlines]);

const chart = fc
    .chartCartesian(d3.scaleTime(), d3.scaleLinear())
    .svgPlotArea(multi)
    .webglPlotArea(candlestick)

chart.xDomain(xExtent(data));
chart.yDomain(yExtent(data));

d3.select('#chart')
    .datum(data)
    .call(chart);
```

Before:

<img width="1771" alt="Screen Shot 2024-01-15 at 3 06 03 PM" src="https://github.com/d3fc/d3fc/assets/143196169/fc25f7e3-cf6c-4e84-b4a7-b194c12b38ff">

After:

<img width="1772" alt="Screen Shot 2024-01-15 at 3 05 42 PM" src="https://github.com/d3fc/d3fc/assets/143196169/b6599e15-4b93-4fa6-88d1-cd89f37cafb6">

dependabot[bot] committed:
chore(deps-dev): bump word-wrap from 1.2.3 to 1.2.4

Bumps [word-wrap](https://github.com/jonschlinkert/word-wrap) from 1.2.3 to 1.2.4.
- [Release notes](https://github.com/jonschlinkert/word-wrap/releases)
- [Commits](https://github.com/jonschlinkert/word-wrap/compare/1.2.3...1.2.4)

---
updated-dependencies:
- dependency-name: word-wrap
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

lincolnbergeson committed:
fix: doji candlesticks in webgl

tradingcage commented:
I see this error in CI but can't figure out what it means. Does it have anything to do with my PR?

```
> commitlint
> commitlint --from HEAD~1 to HEAD --verbose

⧗   input: Merge 943cf15873f236e6e9edca42ade[69](https://github.com/d3fc/d3fc/actions/runs/7534043549/job/20507706889?pr=1804#step:4:70)413fe460610 into 87d2372850b5b0c76bd8d1b09f06078b0a649865
✔   found 0 problems, 0 warnings
⧗   input: Fix doji candlesticks in webgl
✖   subject may not be empty [subject-empty]
✖   type may not be empty [type-empty]
```

tradingcage commented:
@ColinEberhardt What do you think about this fix?

I am planning to make more such contributions to this library over the next year or so as I continue to work on [FirChart](https://github.com/tradingcage/firchart). I'm happy to continue maintaining a fork but would prefer to have things upstream. If you are open to adding a maintainer to this project, let me know and we can discuss what that would look like.

ColinEberhardt commented:
@tradingcage apologies for the slow reply, I've not had much time to work on this library recently. However, more recently help has arrived and we plan to address some of the recent issues and generally bring things back up-to-date. We'll endeavour to get this PR merged shortly, and would appreciate future contributions :-)

jisted-scottlogic committed:
chore: update date

jisted-scottlogic committed:
fix: lerna version compatibility

jisted-scottlogic committed:
fix: lerna version compatibility

jisted-scottlogic committed:
fix: attempt lerna workaround

jisted-scottlogic committed:
fix: attempt lerna workaround

jisted-scottlogic committed:
fix: solve deprecation (lerna)

jisted-scottlogic committed:
fix: solve deprecation (lerna)

jisted-scottlogic committed:
fix: replace lerna bootstrap with npm workspaces

jisted-scottlogic committed:
fix: replace lerna bootstrap with npm workspaces

cfisher-scottlogic committed:
ci: refine CI workflows for operational clarity

cfisher-scottlogic committed:
ci: prevent 'release' job from running on forks

cfisher-scottlogic committed:
docs: add CI section to project README

jisted-scottlogic committed:
fix: pin node to fix gh actions

dependabot[bot] committed:
chore(deps-dev): bump ejs from 3.1.9 to 3.1.10

Bumps [ejs](https://github.com/mde/ejs) from 3.1.9 to 3.1.10.
- [Release notes](https://github.com/mde/ejs/releases)
- [Commits](https://github.com/mde/ejs/compare/v3.1.9...v3.1.10)

---
updated-dependencies:
- dependency-name: ejs
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps-dev): bump @babel/traverse from 7.22.5 to 7.24.5

Bumps [@babel/traverse](https://github.com/babel/babel/tree/HEAD/packages/babel-traverse) from 7.22.5 to 7.24.5.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.24.5/packages/babel-traverse)

---
updated-dependencies:
- dependency-name: "@babel/traverse"
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): yargs-parser, commitlint/cli, commitlint/travis-cli

Bumps [yargs-parser](https://github.com/yargs/yargs-parser) to 21.1.1 and updates ancestor dependencies [yargs-parser](https://github.com/yargs/yargs-parser), [@commitlint/cli](https://github.com/conventional-changelog/commitlint/tree/HEAD/@commitlint/cli) and [@commitlint/travis-cli](https://github.com/conventional-changelog/commitlint/tree/HEAD/@commitlint/travis-cli). These dependencies need to be updated together.

Updates `yargs-parser` from 10.1.0 to 21.1.1
- [Release notes](https://github.com/yargs/yargs-parser/releases)
- [Changelog](https://github.com/yargs/yargs-parser/blob/main/CHANGELOG.md)
- [Commits](https://github.com/yargs/yargs-parser/compare/v10.1.0...yargs-parser-v21.1.1)

Updates `@commitlint/cli` from 8.3.6 to 19.3.0
- [Release notes](https://github.com/conventional-changelog/commitlint/releases)
- [Changelog](https://github.com/conventional-changelog/commitlint/blob/master/@commitlint/cli/CHANGELOG.md)
- [Commits](https://github.com/conventional-changelog/commitlint/commits/v19.3.0/@commitlint/cli)

Updates `@commitlint/travis-cli` from 8.3.6 to 19.3.0
- [Release notes](https://github.com/conventional-changelog/commitlint/releases)
- [Changelog](https://github.com/conventional-changelog/commitlint/blob/master/@commitlint/travis-cli/CHANGELOG.md)
- [Commits](https://github.com/conventional-changelog/commitlint/commits/v19.3.0/@commitlint/travis-cli)

---
updated-dependencies:
- dependency-name: yargs-parser
  dependency-type: indirect
- dependency-name: "@commitlint/cli"
  dependency-type: direct:development
- dependency-name: "@commitlint/travis-cli"
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump dot-prop and @commitlint/config-conventional

Bumps [dot-prop](https://github.com/sindresorhus/dot-prop) to 5.3.0
and updates ancestor dependency
[@commitlint/config-conventional]
These dependencies need to be updated together.

Updates `dot-prop` from 3.0.0 to 5.3.0

Updates `@commitlint/config-conventional` from 8.3.6 to 19.2.2
---
updated-dependencies:
- dependency-name: dot-prop
  dependency-type: indirect
- dependency-name: "@commitlint/config-conventional"
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>

bethpritchard commented:
> I see this error in CI but can't figure out what it means. Does it have anything to do with my PR?
> 
> ```
> > commitlint
> > commitlint --from HEAD~1 to HEAD --verbose
> 
> ⧗   input: Merge 943cf15873f236e6e9edca42ade[69](https://github.com/d3fc/d3fc/actions/runs/7534043549/job/20507706889?pr=1804#step:4:70)413fe460610 into 87d2372850b5b0c76bd8d1b09f06078b0a649865
> ✔   found 0 problems, 0 warnings
> ⧗   input: Fix doji candlesticks in webgl
> ✖   subject may not be empty [subject-empty]
> ✖   type may not be empty [type-empty]
> ```

Hi @tradingcage 

This CI failure is due to commitlint - you need to have commit messages in the form `type: title`, such as `fix: fixed this`. Please see [CommitLint](https://github.com/conventional-changelog/commitlint) for more information

jisted-scottlogic committed:
Merge remote-tracking branch 'upstream/master'

jisted-scottlogic committed:
fix: resolve merge issues

jisted-scottlogic committed:
chore: add changeset

jisted-scottlogic commented:
Closing this one but with important commits cherry-picked onto [this branch](https://github.com/d3fc/d3fc/pull/1870). Thank you. I updated this fork during the process, so please update locally if needed.
